### ISSUE TITLE
onSelectSlot start date is from the start of the day, not where I start the selection (in week view)

### ISSUE THREAD

**Issue by MateiMihaiSTEEP** [jquense/react-big-calendar#2546]
### Check that this is really a bug

- [X] I confirm

### Reproduction link

https://jquense.github.io/react-big-calendar/examples/index.html?path=/docs/props--on-select-slot

### Bug description

- I try to select some slots to create an event(in week view), and it works on the last day of the week, but in the rest the selection starts from the start of the day no matter where I select.
- it can be even reproduced on the oficial link with the examples 

### Expected Behavior

the start of the slot is not from where I placed the mouse(in week view)

### Actual Behavior

when I select a slot(in week view), I want to start from where I first stared selecting, not from another place.

### react-big-calendar version

1.11.2 and 11.11.0

### React version

18.2.0

### Platform/Target and Browser Versions

Chrome, Safari - only here I tested 

### Validations

- [X] Read the [docs](http://jquense.github.io/react-big-calendar/examples).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.
- [X] Make sure this is a react-big-calendar issue and not an implementation issue

### Would you like to open a PR for this bug?

- [ ] I'm willing to open a PR

https://github.com/jquense/react-big-calendar/assets/108053738/e44cca0c-17f2-43fc-95b1-ee59ffe42e78

**Comment by nourbenamor2001** [comment]
I made some simple changes to the code within the repository, the file onSelectSlot.js which handles this functionality:
These are the results:
![image](https://github.com/jquense/react-big-calendar/assets/146457416/8172d883-df9c-4b41-bc63-9992e267da14)

https://github.com/jquense/react-big-calendar/assets/146457416/1255c908-f4df-45ab-ac71-c1c1328615e3

**I tried to modify it by changing the _const onSelectSlot_:**
```
  const onSelectSlot = useCallback((slotInfo) => {
    /**
     * Here we are waiting 250 milliseconds (use what you want) prior to firing
     * our method. Why? Because both 'click' and 'doubleClick'
     * would fire, in the event of a 'doubleClick'. By doing
     * this, the 'click' handler is overridden by the 'doubleClick'
     * action.
     */
    window.clearTimeout(clickRef?.current)
    clickRef.current = window.setTimeout(() => {
      window.alert(buildMessage(slotInfo))
    }, 250)
  }, [])

```
**to this:**
```
const onSelectSlot = useCallback((slotInfo) => {
    /**
     * Here we are waiting 250 milliseconds (use what you want) prior to firing
     * our method. Why? Because both 'click' and 'doubleClick'
     * would fire, in the event of a 'doubleClick'. By doing
     * this, the 'click' handler is overridden by the 'doubleClick'
     * action.
     */
    const { start, end } = slotInfo;
    const adjustedStart = DateTime.fromJSDate(start).startOf('day'); // Ensure start of day

    window.clearTimeout(clickRef?.current);
    clickRef.current = window.setTimeout(() => {
      window.alert(buildMessage({ start: adjustedStart, end })); // Use adjustedStart
    }, 250);
  }, [])

```

**It ends up in this code:**
```
import React, { useCallback, useRef, useEffect, useMemo } from 'react'
import moment from 'moment'
import { Calendar, momentLocalizer } from '../../src'
import demoEvents from '../resources/events'
import mdx from './onSelectSlot.mdx'
import { DateTime } from 'luxon';

const mLocalizer = momentLocalizer(moment)

export default {
  title: 'props',
  component: Calendar,
  parameters: {
    docs: {
      page: mdx,
    },
  },
}

function buildMessage(slotInfo) {
  return `[onSelectSlot] a date selection was made, passing 'slotInfo'
  ${JSON.stringify(slotInfo, null, 2)}`
}


export function OnSelectSlot() {
  const clickRef = useRef(null)

  useEffect(() => {
    /**
     * What Is This?
     * This is to prevent a memory leak, in the off chance that you
     * teardown your interface prior to the timed method being called.
     */
    return () => {
      window.clearTimeout(clickRef?.current)
    }
  }, [])

  const onSelectSlot = useCallback((slotInfo) => {
    /**
     * Here we are waiting 250 milliseconds (use what you want) prior to firing
     * our method. Why? Because both 'click' and 'doubleClick'
     * would fire, in the event of a 'doubleClick'. By doing
     * this, the 'click' handler is overridden by the 'doubleClick'
     * action.
     */
    const { start, end } = slotInfo;
    const adjustedStart = DateTime.fromJSDate(start).startOf('day'); // Ensure start of day

    window.clearTimeout(clickRef?.current);
    clickRef.current = window.setTimeout(() => {
      window.alert(buildMessage({ start: adjustedStart, end })); // Use adjustedStart
    }, 250);
  }, [])

  const defaultDate = useMemo(() => new Date(2015, 3, 1), [])

  return (
    <div className="height600">
      <Calendar
        defaultDate={defaultDate}
        events={demoEvents}
        localizer={mLocalizer}
        onSelectSlot={onSelectSlot}
        selectable
      />
    </div>
  )
}
OnSelectSlot.storyName = 'onSelectSlot'

```

I hope this helps! @IzaanAnwar @MateiMihaiSTEEP

**Comment by forceddd** [comment]
This issue should only exist in WEEK  and DAY. I created a [PR](https://github.com/jquense/react-big-calendar/pull/2545). I hope it can help you.

**Comment by MateiMihaiSTEEP** [comment]
is this change in the 11.1.3 ? because is the same issue

**Comment by MateiMihaiSTEEP** [comment]
the actual problem that I have is in week view, I try to select slots in the same day but it starts always from the beginning go the day not from where I select

**Comment by forceddd** [comment]
> is this change in the 11.1.3 ? because is the same issue

Not yet, this fix has not been merged into the master branch yet

**Comment by Solarkadakiadam** [comment]
Hey everyone, anyone knows which older version i can use to get around this issue? Or when will the fix for this issue will be merged?

**Comment by Solarkadakiadam** [comment]
> > Hey everyone, anyone knows which older version i can use to get around this issue? Or when will the fix for this issue will be merged?
> 
> try version 1.11.1

Have the same error on the 1.11.1 too

**Comment by cutterbl** [comment]
This should now be resolved in v1.11.6

**Comment by Solarkadakiadam** [comment]
> This should now be resolved in v1.11.6

Yes updated the version now its fixed, thank you
